photon-lib/src/test/java/org/photonvision/PhotonPoseEstimatorTest.java

```diff
@@ -526,63 +527,63 @@ void closestToLastPose() {
     @Test
     void pnpDistanceTrigSolve() {
         PhotonCameraInjector cameraOne = new PhotonCameraInjector();
-        PhotonCameraSim cameraOneSim =
-                new PhotonCameraSim(cameraOne, SimCameraProperties.PERFECT_90DEG());
-
         List<VisionTargetSim> simTargets =
                 aprilTags.getTags().stream()
                         .map((AprilTag x) -> new VisionTargetSim(x.pose, TargetModel.kAprilTag36h11, x.ID))
                         .toList();
-
-        /* Compound Rolled + Pitched + Yaw */
-
-        Transform3d compoundTestTransform =
-                new Transform3d(
-                        -Units.inchesToMeters(12),
-                        -Units.inchesToMeters(11),
-                        3,
-                        new Rotation3d(
-                                Units.degreesToRadians(37), Units.degreesToRadians(6), Units.degreesToRadians(60)));
-
-        var estimator =
-                new PhotonPoseEstimator(
-                        aprilTags, PoseStrategy.PNP_DISTANCE_TRIG_SOLVE, compoundTestTransform);
-
-        /* this is the real pose of the robot base we test against */
-        var realPose = new Pose3d(7.3, 4.42, 0, new Rotation3d(0, 0, 2.197));
-        PhotonPipelineResult result =
-                cameraOneSim.process(
-                        1, realPose.transformBy(estimator.getRobotToCameraTransform()), simTargets);
-
-        estimator.addHeadingData(result.getTimestampSeconds(), realPose.getRotation().toRotation2d());
-
-        var estimatedPose = estimator.update(result);
-        var pose = estimatedPose.get().estimatedPose;
-
-        assertEquals(realPose.getX(), pose.getX(), .01);
-        assertEquals(realPose.getY(), pose.getY(), .01);
-        assertEquals(0.0, pose.getZ(), .01);
-
-        /* Straight on */
-
-        Transform3d straightOnTestTransform = new Transform3d(0, 0, 3, new Rotation3d(0, 0, 0));
-
-        estimator.setRobotToCameraTransform(straightOnTestTransform);
-
-        /* Pose to compare with */
-        realPose = new Pose3d(4.81, 2.38, 0, new Rotation3d(0, 0, 2.818));
-        result =
-                cameraOneSim.process(
-                        1, realPose.transformBy(estimator.getRobotToCameraTransform()), simTargets);
-
-        estimator.addHeadingData(result.getTimestampSeconds(), realPose.getRotation().toRotation2d());
-
-        estimatedPose = estimator.update(result);
-        pose = estimatedPose.get().estimatedPose;
-
-        assertEquals(realPose.getX(), pose.getX(), .01);
-        assertEquals(realPose.getY(), pose.getY(), .01);
-        assertEquals(0.0, pose.getZ(), .01);
+        try (PhotonCameraSim cameraOneSim =
+                new PhotonCameraSim(cameraOne, SimCameraProperties.PERFECT_90DEG())) {
```

Comment:
Why did Java change?

Reply:
@mcm001  Two changes here
- I saw a warning because `PhotonCameraSim` implements `Closeable` but we didn't close `cameraOneSim` here
- I added an additional assertion to verify that the position was based on the correct April tag (initially the python tests were failing because the estimator produced a different pose, and I added the assertion to both to narrow down where the bug was)

I'd be happy to revert the first change, and move that to another PR (possibly fixing that warning everywhere).

Reply:
nah we ball